src/atoms/Button/Button.test.js

```diff
@@ -0,0 +1,73 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Button from "./Button";
+
+describe("<Button />", () => {
+  const renderLikeSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        classFromProps="button--like"
+        buttonText="deixe seu like!"
+      />
+    );
+  };
+  const renderSubmitSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        buttonText="Adicionar"
+        classFromProps="button--submit"
+      />
+    );
+  };
```

Comment:
Uma dica é adicionar uma prop para lidar com essa diferença de estilos entre botões, assim você não precisará passar sempre as classes para todos os botões que for usar. O botão deve ser responsável por essa logica

Reply:
Eu geralmente tenho buscar em libs conhecidas de componentes para ver como lidam com botões de diferentes cores, tamanhos, formatos, etc. Exemplo de duas libs:

- Material UI - Button: https://mui.com/material-ui/react-button/
- Chrakra UI - Button: https://chakra-ui.com/docs/components/button/usage#button-variants

Reply:
Nao entendi muito bem.
Eu botei pra passar a classe como Prop, ai no arquivo pela classe estilo X ou Y é aplicado.
Comoassim passar uma prop pra lidar com diferença e nao precisa passar classes sempre? nao consegui visualizar